File: 0344-reverse-string/0344-reverse-string.ts
/**
 Do not return anything, modify s in-place instead.
 */
function reverseString(s: string[]): void {
    var frwd_pointer = 0
    var end_pointer = s.length-1
    while (true){
        if (end_pointer > frwd_pointer){
            var temp = s[end_pointer]; // o
            s[end_pointer] = s[frwd_pointer]; //h at end
            s[frwd_pointer] = temp; //o at start
            frwd_pointer +=1;
            end_pointer -=1;
            console.log(s);
        }else{
            break
        }
    }
};

File: 0344-reverse-string/0344-reverse-string.py
class Solution:
    def reverseString(self, s: List[str]) -> None:
        """
        Do not return anything, modify s in-place instead.
        """
        frwd_pointer = 0
        end_pointer = len(s)-1
        while True:
            if end_pointer > frwd_pointer:
                s[frwd_pointer],s[end_pointer] = s[end_pointer],s[frwd_pointer]
                end_pointer -= 1
                frwd_pointer += 1
            else:
                break
        return s
